services/github/github-release.service.js

```diff
@@ -39,22 +92,106 @@ module.exports = class GithubRelease extends GithubAuthService {
     }
   }
 
-  static render({ version, isPrerelease }) {
-    return {
-      message: addv(version),
-      color: isPrerelease ? 'orange' : 'blue',
+  static render({ version, sort, isPrerelease }) {
+    let color = 'blue'
+    color = sort === 'semver' ? versionColor(version) : color
+    color = isPrerelease ? 'orange' : color
+    return { label: 'release', message: addv(version), color }
+  }
+
+  async fetchReleases({ user, repo }) {
+    return this._requestJson({
+      url: `/repos/${user}/${repo}/releases`,
+      schema: releaseInfoArraySchema,
+      errorMessages: errorMessagesFor('no releases or repo not found'),
+    })
+  }
+
+  static getLatestRelease({ releases, sort, includePrereleases }) {
+    if (sort === 'semver') {
+      const latestRelease = latest(releases.map(release => release.tag_name), {
+        pre: includePrereleases,
+      })
+      const kvpairs = Object.assign(
+        ...releases.map(release => ({ [release.tag_name]: release.prerelease }))
+      )
+      return { tag_name: latestRelease, prerelease: kvpairs[latestRelease] }
+    }
+    if (!includePrereleases) {
+      const stableReleases = releases.filter(release => !release.prerelease)
+      if (stableReleases.length > 0) {
+        return stableReleases[0]
+      }
+      return releases[0]
     }
+    return releases[0]
   }
 
-  async handle({ variant, user, repo }) {
-    const {
-      tag_name: version,
-      prerelease: isPrerelease,
-    } = await fetchLatestRelease(this, {
-      user,
-      repo,
-      includePre: variant === 'release-pre',
+  async handle({ user, repo }, queryParams) {
+    const sort = queryParams.sort
+    const includePrereleases = queryParams.include_prereleases !== undefined
+
+    let latestRelease
+    if (!includePrereleases && sort === 'date') {
+      latestRelease = await fetchLatestRelease(this, { user, repo })
+      return this.constructor.render({
+        version: latestRelease.tag_name,
+        sort,
+        isPrerelease: latestRelease.prerelease,
+      })
+    }
+
+    const releases = await this.fetchReleases({ user, repo })
+    if (releases.length === 0) throw new NotFound({ prettyMessage: 'none' })
+    latestRelease = this.constructor.getLatestRelease({
+      releases,
+      sort,
+      includePrereleases,
+    })
+
+    return this.constructor.render({
+      version: latestRelease.tag_name,
+      sort,
+      isPrerelease: latestRelease.prerelease,
     })
-    return this.constructor.render({ version, isPrerelease })
   }
 }
+
+const redirects = {
+  GithubReleaseRedirect: redirector({
+    name: 'GithubReleaseRedirect',
+    category: 'version',
+    route: {
+      base: 'github/release',
+      pattern: ':user/:repo',
+    },
+    transformPath: ({ user, repo }) => `/github/v/release/${user}/${repo}`,
+    dateAdded: new Date('2019-06-25'),
+  }),
+  GithubReleasePreRedirect: redirector({
+    category: 'version',
```

Comment:
```suggestion
    name: 'GithubReleasePreRedirect',
    category: 'version',    
```

Reply:
I forgot that the `name` would be automatically set so not sure if this is necessary

Reply:
I set the names explicitly on the other two redirects because if I didn't do that, they'll both auto-generate the same name which causes an error. I didn't bother explicitly setting a name here because the auto-generated name is unique.

Reply:
Gotcha, makes sense